homeassistant/components/motionmount/entity.py

```diff
@@ -42,12 +42,45 @@ def __init__(self, mm: motionmount.MotionMount, config_entry: ConfigEntry) -> No
                 (dr.CONNECTION_NETWORK_MAC, mac)
             }
 
+    @property
+    def available(self) -> bool:
+        """Return True if the MotionMount is available (we're connected)."""
+        return self.mm.is_connected
+
+    def update_name(self) -> None:
+        """Update the name of the associated device."""
+        # Check whether we need to update our name
+        if self.device_info and self.device_info[ATTR_NAME] != self.mm.name:
+            device_info = self.device_info
+            device_info[ATTR_NAME] = self.mm.name
+            self._attr_device_info = device_info
+
+            # Also update the name in the device registry if needed
+            mac = format_mac(self.mm.mac.hex())
+            device_registry = dr.async_get(self.hass)
+
+            # Find the device
+            if mac == EMPTY_MAC:
+                device = device_registry.async_get_device(
+                    identifiers={(DOMAIN, self._base_unique_id)}
+                )
+            else:
+                device = device_registry.async_get_device(
+                    connections={(dr.CONNECTION_NETWORK_MAC, mac)}
+                )
```

Comment:
This is not needed, entities have a `device_entry` attribute which points to the device.

Reply:
Thanks for pointing this out. It makes the code much more compact.
I couldn't find any reference to this attribute in the documentation though.